lib/pacemaker/pcmk_verify.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2023 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+#include <crm/cib/internal.h>
+#include <crm/common/output.h>
+#include <crm/common/results.h>
+#include <crm/common/scheduler.h>
+#include <pacemaker-internal.h>
+#include <pacemaker.h>
+
+#include <stdint.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "libpacemaker_private.h"
+
+int
+pcmk__parse_cib(pcmk__output_t *out, const char *cib_source, xmlNodePtr *cib_object)
+{
+    int rc = pcmk_rc_ok;
+
+    crm_info("=#=#=#=#= Getting XML =#=#=#=#=");
+
+    if (cib_source == NULL) {
+        crm_info("Reading XML from: live cluster");
+        rc = cib__signon_query(out, NULL, cib_object);
+
+    }  else if (cib_source[0] == '<') {
+        *cib_object = string2xml(cib_source);
+        if (*cib_object == NULL) {
+            rc = ENODATA;
+        }
+
+    } else if (cib_source[0] == '-') {
+        *cib_object = stdin2xml();
+        if (*cib_object == NULL) {
+            rc = ENODATA;
+        }
+
+    } else {
+        *cib_object = filename2xml(cib_source);
+        if (*cib_object == NULL) {
+            rc = ENODATA;
+        }
+    }
+
+    if (rc == ENODATA) {
+        out->err(out, "Couldn't parse input file");
+    }
```

Comment:
I'm thinking the caller should handle the error -- crm_verify can use g_set_error()